js/components/higher-order/accessibleMenu.js

```diff
@@ -0,0 +1,116 @@
+var React = require('react');
+var ReactDOM = require('react-dom');
+var CONSTANTS = require('../../constants/constants');
+
+var AccessibleMenu = function(ComposedComponent) {
+  return React.createClass({
+
+    componentDidMount: function() {
+      this.onKeyDown = this.onKeyDown.bind(this);
+      this.menuDomElement = ReactDOM.findDOMNode(this.composedComponent);
+
+      if (this.menuDomElement) {
+        this.menuDomElement.addEventListener('keydown', this.onKeyDown);
+      }
+    },
+
+    componentWillUnmount: function() {
+      if (this.menuDomElement) {
+        this.menuDomElement.removeEventListener('keydown', this.onKeyDown);
+      }
+    },
+
+    /**
+     * Keydown event handler. Implements arrow key navigation for menu items.
+     * @private
+     * @param {event} event The keyboard event object.
+     */
+    onKeyDown: function(event) {
+      if (
+        !event.target ||
+        !event.target.hasAttribute(CONSTANTS.KEYBD_FOCUS_ID_ATTR)
+      ) {
+        return;
+      }
+
+      switch (event.key) {
+        case CONSTANTS.KEY_VALUES.ARROW_UP:
+        case CONSTANTS.KEY_VALUES.ARROW_LEFT:
+          event.preventDefault();
+          this.focusOnMenuItemSibling(event.target, false);
+          break;
+        case CONSTANTS.KEY_VALUES.ARROW_DOWN:
+        case CONSTANTS.KEY_VALUES.ARROW_RIGHT:
+          event.preventDefault();
+          this.focusOnMenuItemSibling(event.target, true);
+          break;
+        default:
+          break;
+      }
+    },
+
+    /**
+     * Finds the previous or next sibling of the given menu item and gives it focus.
+     * @private
+     * @param {Element} menuItem The menuItem element whose sibling we want to focus on.
+     * @param {Boolean} useNextSibling Choses the sibling next to menuItem when true and the previous one when false.
+     */
+    focusOnMenuItemSibling: function(menuItem, useNextSibling) {
+      var menuItemsList = [];
+      if (this.menuDomElement) {
+        menuItemsList = this.menuDomElement.querySelectorAll('[' + CONSTANTS.KEYBD_FOCUS_ID_ATTR + ']');
+      }
+      if (!menuItemsList.length) {
+        return;
+      }
+      // Since these elements aren't actually next to each other in the DOM, their position
+      // relative to one another is implied from their tab order, which should be the same as
+      // the one returned by querySelectorAll as long as tabindex is set to 0 (which should be the case).
+      var siblingIndex = this.getMenuItemSiblingIndex(menuItemsList, menuItem, useNextSibling);
+      var menuItem = menuItemsList[siblingIndex];
```

Comment:
Since it is possible to access the property called '-1' of the array, should we do an explicit check for the siblingIndex to be >= 0? The getMenuItemSiblingIndex function currently returns -1 if menuItem is absent from the list.

Reply:
Will add it, but we do check `menuItem` below, so without the check it's a bit cleaner

Reply:
Actually we can leave it as is. Checking menuItem and menuItem.focus is enough